### DISCUSSION TITLE
Is there any way to get context outside build?

### DISCUSSION THREAD

**Discussion by soxft** [rrousselGit/flutter_hooks#443] [Q&A]
Hi, I'm currently working on a project using Flutter_hooks and the windows_manager package. I need to override some methods from WindowListener provided by windows_manager outside of the build method. However, I still need to access the context within these overridden methods.

Could you please provide guidance or a solution on how to achieve this with Flutter_hooks? Specifically, I'm looking for a way to safely access context outside of the build method.

Thank you!

**Comment by benthillerkus** [comment]
> Could you please provide guidance or a solution on how to achieve this with Flutter_hooks?

`flutter_hooks` is pretty much unrelated to your problem and doesn't help or hinder you in any way.

> Specifically, I'm looking for a way to safely access context outside of the build method.

As long as you check if the context is mounted before using it (and after every await), it's safe.


The strategy should be to pick a build context that will live as long as your functionality that requires a build context.
